components/Main/NewsletterSignUp/SignUpForm/index.tsx

```diff
@@ -1,37 +1,22 @@
-import React, { useState } from 'react'
-import style from './SignUpForm.module.css'
-import type { ChangeEvent, Dispatch, FormEvent, SetStateAction } from 'react'
+import Button from '../../Button'
+import Form from '../../form/Form'
+import InputField from '../../form/InputField'
+import type { Dispatch, FormEvent, SetStateAction } from 'react'
 
 export type SignUpFormProps = {
   setIsSuccessful: Dispatch<SetStateAction<boolean>>
 }
 
 export default function SignUpForm({ setIsSuccessful }: SignUpFormProps): JSX.Element {
-  const [email, setEmail] = useState('')
-
-  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setEmail(event.target.value)
-  }
-
   const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
     event.preventDefault()
     setIsSuccessful(true)
   }
 
   return (
-    <form onSubmit={handleSubmit}>
-      <input
-        className={style.input}
-        name="email"
-        onChange={handleChange}
-        placeholder="Enter Email"
-        required
-        type="email"
-        value={email}
-      />
-      <button className={style.button} type="submit">
-        Join
-      </button>
-    </form>
+    <Form onSubmit={handleSubmit}>
+      <InputField name="email" placeholder="Enter Email" required type="email" />
```

Comment:
One thing that appears to be missing from our `InputField` and `TextareaField` components APIs is exposing an `onChange` prop to allow these components to be controlled by the form. Right now, without interacting with the DOM directly, we don't have a way to manage the state of these from the `Form` components.

Reply:
Agreed. Also, how are we handling ARIA attributes?

Reply:
Weird, I would have thought I rename this prop to handleClick. Thanks for explaining. Definitely, will update in the next commit.

Reply:
If there is a label element for the input/textarea component, just label, no aria-label.